x/bank/keeper/grpc_query.go

```diff
@@ -47,10 +43,6 @@ func (k BaseKeeper) AllBalances(ctx context.Context, req *types.QueryAllBalances
 		return nil, status.Error(codes.InvalidArgument, "empty request")
 	}
 
-	if req.Address == "" {
```

Comment:
why these removals

Reply:
this condition is checked again using `sdk.AccAddressFromBech32(req.Address)` in L46 https://github.com/cosmos/cosmos-sdk/pull/14045/files#diff-c2a808829580789ad27e7f3393b3cd63b2f61ff6df06de938cd14dfd73d01e9fR46